server/app/services/question/types/MultiOptionQuestionDefinition.java

```diff
@@ -144,41 +144,25 @@ ImmutableSet<CiviFormError> internalValidate(Optional<QuestionDefinition> previo
 
     OptionalInt minChoicesRequired = getMultiOptionValidationPredicates().minChoicesRequired();
     OptionalInt maxChoicesAllowed = getMultiOptionValidationPredicates().maxChoicesAllowed();
-    if (minChoicesRequired.isPresent()) {
-      if (minChoicesRequired.getAsInt() < 0) {
```

Comment:
**suggestion (blocking):** From my understanding, best practice is to validate twice, once client-side and once server-side, to be certain data is in the right format.  Client-side validation can be easy for malicious actors to bypass.  Also, server-side validation means we can guarantee the API will return valid data. Without it, we can't make that promise.  My suggestion would be to keep this validation and to add server-side validation for the min >= 0 and max is >= 1 on the other types of question configs.

[An article of using both types of validation](https://surveyjs.io/stay-updated/blog/client-server-data-validation#:~:text=In%20general%2C%20it%20is%20recommended,all%20data%20is%20validated%20correctly.)

Reply:
Thanks for the feedback and explanation! I updated all the questions to include server-side validation to the min/max values.